### PR TITLE
Allow to specify colors for text in markup

### PR DESCRIPTION
**Pull request by Gusted**
- Allow users to set colors for their text elements in rendered markup. So `<span style="color: red">Hello World!</span>` will now be accepted by Bluemonday, other properties are still disallowed by Bluemonday.

### CODE REVIEW

**Comment by silverwind** [comment]
Is this same from stuff like `color: url('https://example.com')`?

**Comment by CLanguagePurist** [comment]
> Is this same from stuff like `color: url('https://example.com')`?

Nope, neither Firefox nor Chromium based browser would load URL for color in CSS. I've tested it just now.

Also if you check in this documentation for formal syntax, you'll notice that there is no way to load any external content for color: https://developer.mozilla.org/en-US/docs/Web/CSS/color

So color alone seems safe.

File: modules/markup/sanitizer_test.go [hunk 1]
```diff
@@ -45,6 +45,11 @@ func Test_Sanitizer(t *testing.T) {
 		`<input type="checkbox" disabled=""/>unchecked`, `<input type="checkbox" disabled=""/>unchecked`,
 		`<span class="emoji dropdown">NAUGHTY</span>`, `<span>NAUGHTY</span>`,
 		`<span class="emoji">contents</span>`, `<span class="emoji">contents</span>`,
+
+		// Color property
+		`<span style="color: red">Hello World</span>`, `<span style="color: red">Hello World</span>`,
+		`<p style="color: red">Hello World</p>`, `<p style="color: red">Hello World</p>`,
+		`<code style="color: red">Hello World</code>`, `<code>Hello World</code>`,
```
Comment (wxiaoguang) [inline modules/markup/sanitizer_test.go:52]: Does it need a test for `<span style="non-color: ...."` to cover the negative case?

Comment (Prox501) [reply modules/markup/sanitizer_test.go:52]: Maybe to just verify bluemonday's implementation to ensure that it only allows `color` style property and that the policy is being applied correctly.

Comment (Prox501) [reply modules/markup/sanitizer_test.go:52]: would the following suffice?

```go
		`<span style="bad-color: red">Hello World</span>`, `<span>Hello World</span>`,
		`<p style="bad-color: red">Hello World</p>`, `<p>Hello World</p>`,
		`<code style="bad-color: red">Hello World</code>`, `<code>Hello World</code>`,
```

**Comment by silverwind** [comment]
There is a issue thought that some colors will lead to low-contrast depending on light/dark theme and the won't react to theme changes either.  We could also limit this to just a set of named color strings and [map them to the existing CSS vars](https://github.com/go-gitea/gitea/blob/931c02d1521b7b990f70a590c691b066dbc27d4d/web_src/less/_base.less#L65-L77).

**Review by silverwind — APPROVED** [review]

**Review by wxiaoguang — APPROVED** [review]

**Comment by wxiaoguang** [comment]
bot

**Pull request merged**